invenio_oauthclient/utils.py

```diff
@@ -89,23 +91,22 @@ def oauth_authenticate(client_id, user, require_existing_link=False,
     return False
 
 
-def oauth_register(account_info, form_data=None):
+def oauth_register(form):
     """Register user if possible."""
-    email = account_info.get("email")
-    if form_data and form_data.get("email"):
-        email = form_data.get("email")
-
-    if email:
-        data = dict(**form_data) if form_data else {}
-        data['password'] = ''
-        data['email'] = email
+    data = deepcopy(form.data)
+    form = disable_csrf_form(form)
+
+    if form.validate():
+        if 'password' not in data or not data['password']:
```

Comment:
Just a small code-style advice:
`if 'password' not in data or not data['password']:` == `if not data.get('password')`
:)